flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -134,15 +140,33 @@ public void snapshotState(FunctionSnapshotContext context) throws Exception {
   public void run(SourceContext<FlinkInputSplit> ctx) throws Exception {
     this.sourceContext = ctx;
     while (isRunning) {
+      LOG.debug("Start polling snapshots from snapshot id: {}, max-snapshots-per-monitor-interval: {}", lastSnapshotId,
+          scanContext.maxSnapshotsPerMonitorInterval());
+      long start = System.currentTimeMillis();
+      long startSnapshotId = lastSnapshotId;
       synchronized (sourceContext.getCheckpointLock()) {
         if (isRunning) {
           monitorAndForwardSplits();
         }
       }
+      LOG.debug("Forwarded splits from {}(exclusive) to {}(inclusive), time elapsed {}ms",
```

Comment:
I think it is better to log the durations for the two steps separately if we want to have better understanding of the bottleneck

Reply:
The logging seems wrong to me. both `startSnapshotId` and `lastSnapshotId` are pointing to the same snapshotId. Moving the logging inside `maxReachableSnapshotId`, we can correctly apply the endSnapshotId calculated from `maxReachableSnapshotId`.

Reply:
`lastSnapshotId` is updated in `monitorAndForwardSplits`.

Reply:
yes. I missed `lastSnapshotId`. It also means that the code is difficult to read. we are relying on the side effect of the `monitorAndForwardSplits` method. It is more clear to avoid such side effect. Plus it is better to measure the latency separately: planning vs emitting.

Reply:
@chenjunjiedada I prefer we address the above comment regarding logging

Reply:
Will do.